src/main/java/net/snowflake/ingest/connection/InsertFilesClientInfo.java

```diff
@@ -37,6 +37,9 @@
 
   /** Constructor with both fields as required. */
   public InsertFilesClientInfo(long clientSequencer, String offsetToken) {
+    if (clientSequencer < 0) {
+      throw new IllegalArgumentException("ClientSequencer should be non negative.");
+    }
     this.clientSequencer = clientSequencer;
     this.offsetToken = offsetToken;
```

Comment:
lets file a jira to also do prechecks on offsetToken. We already have some checks on size on SF side because this will be in fdb but good to have in sdk too.

Reply:
That's fine, i'll add them in this pr.